File: WEB + CodenameOne/src/Entity/Vol.php
<?php

namespace App\Entity;

use Doctrine\ORM\Mapping as ORM;
use Symfony\Component\Validator\Constraints as Assert;
use Symfony\Component\Serializer\Annotation\Groups;



/**
 * Vol
 *
 * @ORM\Table(name="vol")
 * @ORM\Entity(repositoryClass="App\Repository\VolRepository")
 */
class Vol
{
    /**
     * @var int

     * @ORM\Column(name="id_vol", type="integer", nullable=false)
     * @ORM\Id
     * @ORM\GeneratedValue(strategy="IDENTITY")
     * @Groups ("post:read")

     */
    private $idVol;

    /**
     * @var string
     * @Assert\NotBlank(message="Ref doit etre non vide")
     * @Assert\Length(
     * max = 5,
     * maxMessage="Longueur maximale est 5",
     * min = 4,
     * minMessage="Longueur minimale est 4"
     * )
     * @ORM\Column(name="ref", type="string", length=20, nullable=false)
     * @Groups ("post:read")
     */
    private $ref;

    /**
     * @var string
     * @Assert\NotBlank(message="DEPART doit etre non vide")
     * @Assert\Length(
     * max = 40, 
     * maxMessage="Longueur maximale est 40"
     
     * )
     * @ORM\Column(name="depart", type="string", length=40, nullable=false)
     *  @Groups ("post:read")
     */
    private $depart;

    /**
     * @var string
     * @Assert\NotBlank(message="DESTINATION doit etre non vide")
     * @Assert\Length(
     * max = 40, 
     * maxMessage="Longueur maximale est 40"
     * )
     * @ORM\Column(name="destination", type="string", length=40, nullable=false)
     *  @Groups ("post:read")
     */
    private $destination;

    /**
     * @var \DateTime
     * @Assert\NotBlank(message="DATE DEPART doit etre non vide")
     * @Assert\GreaterThan("today")
     * @ORM\Column(name="date_d", type="date", nullable=false)
     *  @Groups ("post:read")
     */
    private $dateD;

    /**
     * @var \DateTime
     * @Assert\GreaterThan("today")
     * @Assert\NotBlank(message="DATE RETOUR doit etre non vide")
     * @ORM\Column(name="date_r", type="date", nullable=false)
     *  @Groups ("post:read")
     */
    private $dateR;

    /**
     * @var float
     * @Assert\NotBlank(message="PRIX doit etre non vide")
     * @ORM\Column(name="prix_v", type="float", precision=10, scale=0, nullable=false)
     *  @Groups ("post:read")
     */
    private $prixV;

    public function getIdVol(): ?int
    {
        return $this->idVol;
    }
    public function setIdVol(string $idVol): self
    {
        $this->idVol = $idVol;

        return $this;
    }
    public function getRef(): ?string
    {
        return $this->ref;
    }

    public function setRef(string $ref): self
    {
        $this->ref = $ref;

        return $this;
    }

    public function getDepart(): ?string
    {
        return $this->depart;
    }

    public function setDepart(string $depart): self
    {
        $this->depart = $depart;

        return $this;
    }

    public function getDestination(): ?string
    {
        return $this->destination;
    }

    public function setDestination(string $destination): self
    {
        $this->destination = $destination;

        return $this;
    }

    public function getDateD(): ?\DateTimeInterface
    {
        return $this->dateD;
    }

    public function setDateD(\DateTimeInterface $dateD): self
    {
        $this->dateD = $dateD;

        return $this;
    }

    public function getDateR(): ?\DateTimeInterface
    {
        return $this->dateR;
    }

    public function setDateR(\DateTimeInterface $dateR): self
    {
        $this->dateR = $dateR;

        return $this;
    }

    public function getPrixV(): ?float
    {
        return $this->prixV;
    }

    public function setPrixV(float $prixV): self
    {
        $this->prixV = $prixV;

        return $this;
    }


}

File: DESKTOP/src/entities/Vol.java
/*
 * Click nbfs://nbhost/SystemFileSystem/Templates/Licenses/license-default.txt to change this license
 * Click nbfs://nbhost/SystemFileSystem/Templates/Classes/Class.java to edit this template
 */
package entities;
import java.sql.Date;
/**
 *
 * @author hp
 */
public class Vol {
    private int id_vol; 
    private double prix_v;
    private String depart,destination,ref;
    private Date date_d,date_r;
 

   public Vol () {
       
   }

    public Vol(int id_vol,String ref,String depart, String destination, java.sql.Date date_d, java.sql.Date date_r, double prix_v) {
        this.id_vol = id_vol;
        this.ref = ref;
        this.depart = depart;
        this.destination = destination;
        this.date_d = date_d;
        this.date_r = date_r;
        this.prix_v= prix_v;
    }

    
    

    
   
    @Override
    public String toString() {
        return "Vol{" + "id_vol=" + id_vol + ", ref=" + ref  + ", depart=" + depart + ", destination=" + destination + ", date_d =" + date_d + ", date_r =" + date_r  + ", prix_v =" + prix_v  + '}';
    }

    public int getId_vol() {
        return id_vol;
    }

    public void setId_vol(int id_vol) {
        this.id_vol = id_vol;
    }

    public double getPrix_v() {
        return prix_v;
    }

    public void setPrix_v(double prix_v) {
        this.prix_v = prix_v;
    }

    public String getDepart() {
        return depart;
    }

    public void setDepart(String depart) {
        this.depart = depart;
    }

    public String getDestination() {
        return destination;
    }

    public void setDestination(String destination) {
        this.destination = destination;
    }

    public String getRef() {
        return ref;
    }

    public void setRef(String ref) {
        this.ref = ref;
    }

    public Date getDate_d() {
        return date_d;
    }

    public void setDate_d(Date date_d) {
        this.date_d = date_d;
    }

    public Date getDate_r() {
        return date_r;
    }

    public void setDate_r(Date date_r) {
        this.date_r = date_r;
    }

    
   

    
}
